akka-actor/src/main/scala/akka/actor/dungeon/DeathWatch.scala

```diff
@@ -11,7 +11,7 @@ import akka.event.AddressTerminatedTopic
 
 private[akka] trait DeathWatch { this: ActorCell ⇒
 
-  private var watching: Set[ActorRef] = ActorCell.emptyActorRefSet
+  private var watching: Map[ActorRef, Option[Any]] = Map.empty
```

Comment:
could maybe be a value class, but otherwise we must be careful to not grow the byte size of the actor cell